assets/js/ajax-filters.js

```diff
@@ -255,26 +262,67 @@ jQuery( document ).ready( function ( $ ) {
 		$supports_html5_history = false;
 	}
 
-	var location = document.location.href.split('#')[0];
-
 	function job_manager_store_state( target, page ) {
+		// changed this to ? because this is the standard query string identifier in php
+		// and changed to window.location for better cross browser support
+		// and moved into the function so it rechecks everytime the function is called
+		var location = window.location.href.split('?')[0];
+		var query = window.location.href.split('?')[1];
+		query = query.split('&');
```

Comment:
Ok so thought about this and the real issue is that we need to isolate GET variables specific to WP job manager. Using ? parameters does not affect WP_Query as per the [final comment on this thread](https://core.trac.wordpress.org/ticket/16373). Its when we push or replace HTML5 window.history on circa line [302](https://github.com/lindsaymacvean/WP-Job-Manager/blob/master/assets/js/ajax-filters.js#L302) that the problem occurs because this may confuse the WP engine by removing WP query parameters necessary to direct to the correct page.

Whether we switch back to a # or not does not change the fundamental challenge of isolating just the three parameters relevant to our api, using them and if necessary adding updated parameters into the HTML5 history all while keeping any other parts of the complete URL string intact, both before and after the parameters relevant to us.

I will write something to update this.


[1](http://superuser.com/questions/498617/does-an-anchor-tag-come-before-the-query-string-or-after)